src/hooks/safe-apps/useSafeAppFromManifest.ts

```diff
@@ -0,0 +1,26 @@
+import { useEffect } from 'react'
+import { SafeAppData } from '@gnosis.pm/safe-react-gateway-sdk'
+import { Errors, logError } from '@/services/exceptions'
+import { fetchSafeAppFromManifest } from '@/services/safe-apps/manifest'
+import useAsync from '../useAsync'
+
+type UseSafeAppFromManifestReturnType = {
+  safeApp?: SafeAppData
+  isLoading: boolean
+}
+
+const useSafeAppFromManifest = (appUrl: string, chainId: string): UseSafeAppFromManifestReturnType => {
+  const [data, error, loading] = useAsync<SafeAppData>(
+    () => fetchSafeAppFromManifest(appUrl, chainId),
+    [appUrl, chainId],
+  )
+
+  useEffect(() => {
+    if (!error) return
+    logError(Errors._900, `${appUrl}, ${(error as Error).message}`)
+  }, [appUrl, error])
+
+  return { safeApp: data, isLoading: loading }
```

Comment:
You should probably memoize the object, otherwise it will be returning a new one on each call, potentially causing a re-rendering.

```
return useMemo(() => ({ safeApp: data, isLoading: loading }), [data, loading])
```

Reply:
mmm, I can see it done in several places. Do you think is necessary to memoize return values from hooks instead acting in the component itself when necessary?. Asking because I never used this pattern 🙂 ...

Reply:
I mean, as long as the component restructures it like so `const { safeApp, isLoading }`, it should be fine.
I'll resolve this.

Reply:
aah, now i see. Yep is used like that